keras_core/optimizers/base_optimizer.py

```diff
@@ -329,6 +329,24 @@ def learning_rate(self, learning_rate):
                 )
             self._learning_rate.assign(learning_rate)
 
+    def set_weights(self, weights):
+        """Set the weights of the optimizer."""
+        if not getattr(self, "_built", False):
+            raise ValueError(
```

Comment:
Please add tests for the 2 error messages

Reply:
Added a test in optimizer_test, fixed.